app/jobs/preassembly_job.rb

```diff
@@ -6,8 +6,13 @@ class PreassemblyJob < ApplicationJob
   # @param [JobRun] job_run
   def perform(job_run)
     job_run.started
-    bc = job_run.batch_context
-    result = bc.batch.run_pre_assembly
-    result ? job_run.completed : job_run.completed_with_errors
+    batch = job_run.batch_context.batch
+    batch.run_pre_assembly
+    if batch.had_errors
+      job_run.error_message = batch.error_message
+      job_run.completed_with_errors
+    else
+      job_run.completed
+    end
   end
```

Comment:
this changes around how we report back errors to the job ... instead of returning a boolean from `batch#run_pre_assembly` to indicate errors/no errors, use a new accessor on the `batch` object to indicate if we had errors, and then fetch the error message from the object to store in the job_run if so